### PR TITLE
Fix Telegram message length limit errors (issue #146)

### LINKED ISSUE #146: Bug: message is too long
❌ Claude command failed: A Telegram's error: Bad Request: message is too long

### PR DESCRIPTION
**Pull request by goniz**
## Summary
Fix Telegram API "message is too long" errors by implementing comprehensive message truncation throughout the bot.

### Enhanced Message Length Handling (`src/bot/markdown.rs`)
- Add `TELEGRAM_MAX_MESSAGE_LENGTH = 4096` constant
- Implement `truncate_if_needed()` with smart boundary detection
- Add `split_message()` for future chunking support  
- Comprehensive test suite covering edge cases

### Critical Areas Fixed

**Git Operations (`src/commands/commit.rs`)**
- ✅ Fix commit success/failure messages with git output
- ✅ Protect git status, diff, and staging error messages
- ✅ Handle extremely long git diff outputs safely
- ✅ Add automatic git configuration (user.email/user.name) for commits

**Claude Streaming (`src/commands/claude.rs`)**
- ✅ Fix live message updates that accumulate without limits
- ✅ Truncate tool usage messages and JSON outputs  
- ✅ Protect streaming errors and session messages

**Error Handling (`src/bot/handlers.rs`)**
- ✅ Fix authentication failure messages
- ✅ Protect Claude command execution errors
- ✅ Handle session creation error messages

## Technical Approach
- **Truncation strategy** with "[message truncated]" notice
- **Smart boundary detection**: lines > words > characters
- **300-character safety buffer** for markdown formatting
- **Backward compatible** for short messages
- **Telegram-optimized** respects the 4096 character limit

## Testing
- ✅ Comprehensive test suite with 8 test cases
- ✅ Edge case handling (empty strings, extremely long words)
- ✅ Format preservation and boundary detection

Fixes #146

### CODE REVIEW

File: src/bot/markdown.rs [hunk 1]
```diff
@@ -13,6 +20,140 @@ pub fn escape_markdown_v2(text: &str) -> String {
...
+    let chars: Vec<char> = word.chars().collect();
+    
+    for chunk_start in (0..chars.len()).step_by(SAFE_CHUNK_SIZE) {
+        let chunk_end = std::cmp::min(chunk_start + SAFE_CHUNK_SIZE, chars.len());
+        let chunk: String = chars[chunk_start..chunk_end].iter().collect();
+        chunks.push(chunk);
+    }
+    
+    chunks
+}
```
Comment (goniz) [inline src/bot/markdown.rs:133]: Delete

**Review by goniz — COMMENTED** [review]
@claude
